ablySpec/RealtimeClientChannel.swift

```diff
@@ -607,6 +607,55 @@ class RealtimeClientChannel: QuickSpec {
                     expect(Test.counter).toEventually(equal(2), timeout: testTimeout)
                 }
 
+                context("message cannot be decoded or decrypted") {
+
+                    // RTL7e
+                    it("should deliver with encoding attribute set indicating the residual encoding and error should be emitted") {
+                        let options = AblyTests.commonAppSetup()
+                        options.autoConnect = false
+                        let client = ARTRealtime(options: options)
+                        client.setTransportClass(TestProxyTransport.self)
+                        client.connect()
+                        defer { client.close() }
+
+                        let channelOptions = ARTChannelOptions(encrypted: ARTCrypto.defaultParams())
```

Comment:
Nope, JS is not generating a random key.

@tcard Crypto is meant to be part of the public API, see http://docs.ably.io/client-lib-development-guide/features/#TZ3.  

The pseudo code we'd need for this to be consistent would be:
```
cipherParams = ARTCrypto.defaultParams()
cipherParams.key = randomString()
channelOptions = ARTChannelOptions(encrypted: true, cipherParams: cipherParams)
```

So this needs to be fixed as far as I can tell i.e. ARTCrypto, if public, should not generate a random key ever